### PR TITLE
Refactor Structure

### PR DESCRIPTION
**Pull request by JiejayLan**
### Overview of PR
[reference](https://blog.bitsrc.io/structuring-a-react-project-a-definitive-guide-ac9a754df5eb) I follow to refactor
- [x] remove some unnecessary files
- [x] refactor folder structure
- [x] add dummy tests to increase test coverage
- [x] add react test library [reference](https://testing-library.com/docs/react-testing-library/intro) [reference2](https://github.com/kentcdodds/react-testing-library-examples)

### Architecture Changes
- [x] only keep error and home route

### CODE REVIEW

File: src/routes/AppRouter.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { Route, BrowserRouter as Router, Switch } from 'react-router-dom';
-import ErrorPage from '../pages/ErrorPage';
-import HomePage from '../pages/HomePage';
+import ErrorPage from '../pages/ErrorPage/ErrorPage';
+import HomePage from '../pages/HomePage/HomePage';
```
Comment (ethen001) [inline src/routes/AppRouter.js]: The previous import format is more correct. If you add `export * from './ErrorPage` to an `index.js` file inside the component folder, you don't need to import twice. `import ErrorMessage from '../pages/HomePage ` is enough.

Comment (ethen001) [reply src/routes/AppRouter.js]: Also let's avoid using `export default` to take full advantage of ES6 named imports to reduce imports. Please follow this resources.
[Link 1](https://blog.bitsrc.io/structuring-a-react-project-a-definitive-guide-ac9a754df5eb)
[Link 2](https://stackoverflow.com/questions/29722270/is-it-possible-to-import-modules-from-all-files-in-a-directory-using-a-wildcard/31402500#31402500)

Comment (JiejayLan) [reply src/routes/AppRouter.js]: I agree

File: .eslintignore [hunk 2]
```diff
@@ -1,2 +1 @@
-# all test files
-*.test.js
\ No newline at end of file
+/coverage
```
Comment (songrenzhao) [inline .eslintignore]: EOL, please fix

File: src/components/Message/__tests__/message.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
-import ErrorPage from '../ErrorPage';
+import Message from '../Message';
 
 it('renders without crashing', () => {
```
Comment (songrenzhao) [inline src/components/Message/__tests__/message.js]: Please provide a test suite by doing the following
```js
describe('message test suite', () => {
  it('should render without crashing', async () => {
    ...
  })
})
```

Also please read [Guide](https://devhints.io/jest)

File: src/pages/error/__tests__/error.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import Error from '../Error';
+
+it('renders without crashing', () => {
```
Comment (songrenzhao) [inline src/pages/error/__tests__/error.js]: Please see above

File: src/pages/home/__tests__/home.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import Home from '../Home';
+
+it('renders without crashing', () => {
```
Comment (songrenzhao) [inline src/pages/home/__tests__/home.js]: Please see above

File: src/routes/__tests__/appRouter.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import AppRouter from '../AppRouter';
+
+it('renders without crashing', () => {
```
Comment (songrenzhao) [inline src/routes/__tests__/appRouter.js]: Please see above

**Pull request merged**